### DISCUSSION TITLE
Is it safe?

### DISCUSSION THREAD

**Discussion by istituzionealsole** [Pustur/whatsapp-chat-parser-website#28] [Q&A] [answered]
Hi, the website works great, but I was wondering if the data uploaded there remains on my local broswer session or will be sent somewhere?

I'm worried about the privacy and since I am not a programmer am I unable to verify the code. Thank you

**Comment by Pustur** [comment, marked as answer]
Hi @francescoavagliano, that's a valid concern so thanks for asking.

As stated in the readme:

> The app runs locally and no logs are sent or stored anywhere.

All the data processing of the chat logs happens directly in your browser.

If you want to be extra precautious you can disable your internet connection before uploading the chat log and re-enable it only when you have closed the page (do this in incognito mode)
